crates/core/src/types/type.rs

```diff
@@ -0,0 +1,118 @@
+use crate::prelude::types::*;
+use crate::prelude::YarnFnRegistry;
+use paste::paste;
+use std::any::Any;
+use std::fmt::Debug;
+
+/// All types in the virtual machine, both built-in, i.e. usable in yarn scripts, and internal.
+///
+/// ## Implementation Notes
+///
+/// This type does not exist in the original implementation and was a added as a more idiomatic
+/// representation of the types than dynamic dispatch.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Type {
+    Any(AnyType),
+    Boolean(BooleanType),
+    Function(FunctionType),
+    Number(NumberType),
+    String(StringType),
+}
+
+/// Defines properties that describe a types in the Yarn language.
+///
+/// ## Implementation Notes
+///
+/// - Represents the `IType` interface in the original implementation.
+/// - `Parent` is not implemented because it is set to `AnyType` everywhere anyways.
+///
+pub trait TypeProperties: Clone + PartialEq + Eq + Debug {
+    /// The name of this types.
+    const NAME: &'static str;
+
+    /// A more verbose description of this types.

```

Comment:
I am not sure why types is everywhere. Am I missing something?

Reply:
Lel CLion did this as a helpful refactor when I renamed a module from type to types

Reply:
Helpful Jetbrains, as usual! :D